tests/rustdoc-ui/unescaped_backticks.rs

```diff
@@ -0,0 +1,309 @@
+#![deny(rustdoc::unescaped_backticks)]
+#![allow(rustdoc::broken_intra_doc_links)]
+#![allow(rustdoc::invalid_html_tags)]
+
+/// `
+//~^ ERROR unescaped backtick
+pub fn single() {}
+
+/// \`
+pub fn escaped() {}
+
+/// \\`
+//~^ ERROR unescaped backtick
+pub fn not_escaped() {}
+
+/// \\\`
+pub fn not_not_escaped() {}
+
+/// [`link1]
+//~^ ERROR unescaped backtick
+pub fn link1() {}
+
+/// [link2`]
+//~^ ERROR unescaped backtick
+pub fn link2() {}
+
+/// [`link_long](link_long)
+//~^ ERROR unescaped backtick
+pub fn link_long() {}
+
+/// [`broken-link]
+//~^ ERROR unescaped backtick
+pub fn broken_link() {}
+
+/// <xx:`>
+pub fn url() {}
+
+/// <x:`>
+//~^ ERROR unescaped backtick
+pub fn not_url() {}
```

Comment:
What's the difference between these cases?

Reply:
The first one is a [markdown autolink](https://spec.commonmark.org/0.30/#autolinks) with the url `` xx:` ``, the second one is a regular [text node](https://spec.commonmark.org/0.30/#textual-content) with the text `` <x:`> ``. This happens, because a [url scheme](https://spec.commonmark.org/0.30/#scheme) needs at least two characters.

A lot of this test file has just become a markdown parser test, let me know if I should shorten it.